src/dbus/common/types.hpp

```diff
@@ -517,6 +519,61 @@ struct TxtEntry
     std::vector<uint8_t> mValue;
 };
 
+enum SrpServerState : uint8_t
+{
+    OTBR_SRP_SERVER_STATE_DISABLED = 0, ///< The SRP server is disabled.
+    OTBR_SRP_SERVER_STATE_RUNNING  = 1, ///< The SRP server is running.
+    OTBR_SRP_SERVER_STATE_STOPPED  = 2, ///< The SRP server is stopped.
+};
+
+static_assert(OTBR_SRP_SERVER_STATE_DISABLED == static_cast<uint8_t>(OT_SRP_SERVER_STATE_DISABLED),
```

Comment:
it works for enum with small number of values. 

What we do if the enum have tens of values?

Reply:
One possible way is that we define `OTBR_*` enums by applying a macro function on `OT_*` enums. In this way we can ensure that every `OTBR_*` enum has the same value as the according `OT_*` enum.
```
enum {
    DEFINE_OTBR_ENUM(A),
    DEFINE_OTBR_ENUM(B),
    ...
};
```